### ISSUE TITLE
Race condition between GCObject creation and GC garbage collecting

### ISSUE THREAD

**Issue by mihacooper** [loud-hound/woofer#59]
For example:
**stored-object.cpp:**
```cpp
  SharedTable table = GC::instance().create<SharedTable>();
  ...
  return std::make_unique<GCObjectHolder<SharedTable>>(table.handle());
```
When we go out of scope of this function the `table` which currently is the only one hard link to the new GCObject will be destroyed and GC may collect this object before we start use it.

**Comment by mihacooper** [comment]
done
